src/cutadapt/config.vsh.yaml

```diff
@@ -0,0 +1,446 @@
+          description: Change negative quality values to zero.
+
+    ####################################################################
+    - name: Filtering of processed reads
+      description: |
+        Filters are applied after above read modifications. Paired-end reads are
+        always discarded pairwise (see also --pair_filter).
+      arguments:
+        - name: --minimum_length
+          alternatives: [-m]
+          type: string
+          description: |
+            Discard reads shorter than LEN.
+          default: "0"
+        - name: --maximum_length
+          alternatives: [-M]
+          type: string
+          description: |
+            Discard reads longer than LEN. Default: no limit
+        - name: --max_n
+          type: string
+          description: |
+            Discard reads with more than COUNT 'N' bases. If COUNT is
+            a number between 0 and 1, it is interpreted as a fraction
+            of the read length.
+        - name: --max_expected_errors
+          alternatives: [--max_ee]
+          type: long
+          description: |
+            Discard reads whose expected number of errors (computed
+            from quality values) exceeds ERRORS.
+        - name: --max_average_error_rate
+          alternatives: [--max_aer]
+          type: long
+          description: |
+            as --max_expected_errors (see above), but divided by
+            length to account for reads of varying length.
+        - name: --discard_trimmed
+          alternatives: [--discard]
+          type: boolean_true
+          description: |
+            Discard reads that contain an adapter. Use also -O to
+            avoid discarding too many randomly matching reads.
+        - name: --discard_untrimmed
+          alternatives: [--trimmed_only]
+          type: boolean_true
+          description: |
+            Discard reads that do not contain an adapter.
+        - name: --discard_casava
+          type: boolean_true
+          description: |
+            Discard reads that did not pass CASAVA filtering (header
+            has :Y:).
+
+    ####################################################################
+    - name: Output parameters
+      arguments:
+        - name: --report
+          type: string
+          choices: [full, minimal]
+          description: |
+            Which type of report to print: 'full' or 'minimal'.
+          default: full
+        - name: --json
+          type: boolean_true
+          description: |
+            Write report in JSON format to report.json in
+            the output directory.
+        - name: --output
+          type: file
+          description: |
+            Write trimmed reads to this directory and name the files using {name}.
+            FASTQ or FASTA format is chosen depending on input.
+            Summary report is sent to standard output.
+          default: output
+          direction: output
+          required: true
+          must_exist: true
```

Comment:
Can we split up the output to separate output files?

[CONTRIBUTING.md](https://github.com/viash-hub/biobase/blob/main/CONTRIBUTING.md#step-8-add-arguments-for-the-output-files):

> ## Step 8: Add arguments for output files
> ...
> Note:
>    Preferably, these outputs should not be directores but files. For example, if a tool outputs a directory foo/ containing files foo/bar.txt and foo/baz.txt, there should be two output arguments --bar and --baz (as opposed to one output argument which outputs the whole foo/ directory).

Especially since `cutadapt` also doesn't rely in an output directory,

```
    ${par_json:+--json "${par_output}/report.json"} \
    --output "$par_output/{name}_R1_001.fastq" \
    --paired-output "$par_output/{name}_R2_001.fastq" \
```

Would need to be changed to

```
    ${par_json:+--json "${par_json}"} \
    --output "$par_output" \
    ${par_output_r2:+--paired-output "$par_output_r2"} \
```

or something to that extend

Reply:
Cutadapt implicitly creates a directory because it generates a number of output fastq files depending on the input data and the arguments provided. Hence the `{name}` _variable_. I would like to avoid that users have to set those 'variable' output paths explicitly.

Reply:
I agree that it would be cumbersome if the users would have to set the variable output paths explicitly. I now understand that cutadapt returns a variable number of outputs depending on how many adapters the user provides.

I'm still a bit on the fence because creating one big directory with all of the outputs makes it harder for downstream components to process the files.

Can we set something up with `type: file, multiple: true` such that, by default, the user can use `--output` to get all of the outputs in one directory, but if they provide paths with wildcards the fasta's / fastq's will be written to those paths instead?

* --output_dir: contains all of the files below
* --report_txt: report.txt
* --report_json: report.json
* --output: output_*_R1.fasta/q if paired, else output_*.fasta/q
* --output_r2: output_*_R2.fasta/q

Reply:
I just remembered that `type: file, multiple: true, direction: output` is currently broken and will be fixed in https://github.com/viash-io/viash/pull/639. Will try to finish this PR ASAP.

Reply:
It's not really blocking I think because the bug appears in the VDSL3 module.

Reply:
Coming back to your suggestion, @rcannood:

I have a version in development that does what you suggest: take both an `--output_dir` as well as an `--output` argument. The first is just one directory, the second is `multiple: true`. This allows me to, for instance, run this in the tests:

```bash
"$meta_executable" \
  --output_dir out_test1 \
  --output out_test1/1_001.fasta \
  --output out_test1/unknown_001.fasta \
  ...
```

Or, alternatively `--output out_test1/1_001.fasta;out_test1/unknown_001.fasta`.

I agree it's nice to be able to capture the output explicitly rather than a directory, but it requires one to know the result of the `cutadapt` step prior to running it which will depend on the case at hand.

Ideally though, I would prefer to do something like:

```bash
"$meta_executable" \
  --output_dir out_test1 \
  --output out_test1/*.fast? \
  ...
```

This, however, does not work because of the way the executable is built (`set -f` disallows globbing):

```bash
# check whether required files exist
if [ ! -z "$VIASH_PAR_OUTPUT" ]; then
  IFS=';'
  set -f
  for file in $VIASH_PAR_OUTPUT; do
    unset IFS
    if [ ! -e "$file" ]; then
      ViashError "Output file '$file' does not exist."
      exit 1
    fi
  done
  set +f
fi
```

BTW: The way we deal with this in a Nextflow pipeline is by adding a `map` and glob inside the `file()` function of Nextflow.

Now, how to proceed?

1. Look into adapting the behaviour of Viash? (`-set -f`)
2. Stick to what works with the individual output as illustrated above?
3. Just output the directory name and omit the individual output files?
4. (1) __and__ (2)?